[CORL-1155] Fix theming for rebranded stream/admin

nick-funk opened the pull request:
Props to @cvle for all his hard work peer coding these changes!

## What does this PR do?

Fixes multiple issues with css variables and component duplication to allow theming within the stream.
Also consolidates all of the v1/v2 variables together as well as eliminating the v1 components, we now only have v2 and v3 components.

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

tessalt commented on src/core/client/admin/routes/ForgotPassword/ForgotPasswordForm.tsx:
```diff
@@ -98,8 +98,8 @@ const ForgotPasswordForm: FunctionComponent<Props> = (props) => {
                   <Button
-                    variant="filled"
-                    color="primary"
+                    variant="regular"
```
"regular" is the default for size/variant/color.

nick-funk committed:
Repair theming and CSS variables for the rebrand

This separates the CSS variables into three sections
for admin, stream, as well as a shared set between
them.

This also adds injection of the raw colours from the
design team into post css variables.

Lastly, this re-maps the existing theme variables to
a new consolidated set with fallbacks for compatibility.

Since this is a singular commit from various WIP commits,
I am crediting @cvle as well as myself for this work.

Thank you all who helped make this rebrand with theming
possible!

CORL-1155

Co-authored-by: Chi Vinh Le <vinh@vinh.tech>

nick-funk committed:
Merge remote-tracking branch 'origin/master' into CORL-1155

# Conflicts:
#	src/core/client/admin/components/ModalBodyText.css
#	src/core/client/admin/components/ModalHeader.css
#	src/core/client/admin/components/ModalHeaderUsername.css
#	src/core/client/admin/components/ModerateCard/ApproveButton.css
#	src/core/client/admin/components/ModerateCard/FlagDetailsCategory.css
#	src/core/client/admin/components/ModerateCard/FlagDetailsContainer.css
#	src/core/client/admin/components/ModerateCard/FlagDetailsEntry.css
#	src/core/client/admin/components/ModerateCard/Markers.css
#	src/core/client/admin/components/ModerateCard/ModerateCard.css
#	src/core/client/admin/components/ModerateCard/RejectButton.css
#	src/core/client/admin/components/UserRole/SiteModeratorModal.css
#	src/core/client/admin/components/UserStatus/SuspendForm.css
#	src/core/client/admin/components/UserStatus/SuspendModal.css
#	src/core/client/admin/routes/Community/InviteUsers/RoleField.css
#	src/core/client/ui/components/AbsoluteTime/AbsoluteTime.tsx
#	src/core/client/ui/hooks/useDateTimeFormat.ts

alinhle committed:
Convert old css variables in CallOut

wyattjoh commented on CSS_VARIABLES.md:
```diff
@@ -0,0 +1,295 @@
+
+Color palette that is used for indicating something negative and is usually red.
+
+`--palette-negative-100: #FCE5D9;  /* Before 6.3.0: --palette-error-lightest */`
```
The use of `negative` and `positive` is inconsistent with existing verbiage around colours. The proximity to numbers is especially confusing (i.e.`negative 100`).

Please revert to `error`/`success` naming.

nick-funk committed:
Merge remote-tracking branch 'origin/master' into CORL-1155

# Conflicts:
#	src/core/client/framework/lib/validation.tsx
#	src/core/client/stream/tabs/Comments/Comment/ReplyCommentForm/ReplyCommentForm.css
#	src/core/client/stream/tabs/Comments/Comment/ReplyCommentForm/ReplyCommentForm.tsx
#	src/core/client/stream/tabs/Comments/RTE/RTE.css
#	src/core/client/stream/tabs/Comments/RTE/__snapshots__/RTE.spec.tsx.snap
#	src/core/client/stream/tabs/Comments/Stream/PostCommentForm/PostCommentForm.tsx
#	src/core/client/stream/test/comments/stream/__snapshots__/editComment.spec.tsx.snap
#	src/core/client/ui/components/v2/Button/Button.css
#	src/core/client/ui/components/v2/TextField/TextField.css

nick-funk committed:
Rename postive/negative => success/error

CORL-1155

alinhle committed:
Change some CSS Variable names

alinhle committed:
fix TextField/PasswordField colors
